contracts/Comptroller/Diamond/facets/MarketFacet.sol

```diff
@@ -147,69 +154,122 @@ contract MarketFacet is IMarketFacet, FacetBase {
      * @notice Unlist a market by setting isListed to false
      * @dev Checks if market actions are paused and borrowCap/supplyCap/CF are set to 0
      * @param market The address of the market (vToken) to unlist
-     * @return uint256 0=success, otherwise a failure. (See enum Error for details)
+     * @return uint256 Always NO_ERROR
+     * @custom:event MarketUnlisted is emitted on success
+     * @custom:error MarketNotListed error is thrown when the market is not listed
+     * @custom:error BorrowActionNotPaused error is thrown if borrow action is not paused
+     * @custom:error MintActionNotPaused error is thrown if mint action is not paused
+     * @custom:error RedeemActionNotPaused error is thrown if redeem action is not paused
+     * @custom:error RepayActionNotPaused error is thrown if repay action is not paused
+     * @custom:error EnterMarketActionNotPaused error is thrown if enter market action is not paused
+     * @custom:error LiquidateActionNotPaused error is thrown if liquidate action is not paused
+     * @custom:error BorrowCapIsNotZero error is thrown if borrow cap is not zero
+     * @custom:error SupplyCapIsNotZero error is thrown if supply cap is not zero
+     * @custom:error CollateralFactorIsNotZero error is thrown if collateral factor is not zero
      */
     function unlistMarket(address market) external returns (uint256) {
         ensureAllowed("unlistMarket(address)");
 
         Market storage _market = markets[market];
 
         if (!_market.isListed) {
-            return fail(Error.MARKET_NOT_LISTED, FailureInfo.UNLIST_MARKET_NOT_LISTED);
+            revert MarketNotListed();
         }
 
-        require(actionPaused(market, Action.BORROW), "borrow action is not paused");
-        require(actionPaused(market, Action.MINT), "mint action is not paused");
-        require(actionPaused(market, Action.REDEEM), "redeem action is not paused");
-        require(actionPaused(market, Action.REPAY), "repay action is not paused");
-        require(actionPaused(market, Action.ENTER_MARKET), "enter market action is not paused");
-        require(actionPaused(market, Action.LIQUIDATE), "liquidate action is not paused");
-        require(actionPaused(market, Action.SEIZE), "seize action is not paused");
-        require(actionPaused(market, Action.TRANSFER), "transfer action is not paused");
-        require(actionPaused(market, Action.EXIT_MARKET), "exit market action is not paused");
+        if (!actionPaused(market, Action.BORROW)) {
+            revert BorrowActionNotPaused();
```

Comment:
`ActionNotPaused(Action.BORROW)` to decrease the contract size?

Reply:
[107ead8](https://github.com/VenusProtocol/venus-protocol/pull/604/commits/107ead82d2262f6c754da04be64904dc9f91856e)